file/readfile.go

```diff
@@ -91,13 +91,19 @@ func readContent(reader io.Reader) (*Content, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "validating file content")
 	}
-	err = yaml.Unmarshal(bytes, &content)
+	err = yamlUnmarshal(bytes, &content)
 	if err != nil {
 		return nil, err
 	}
 	return &content, nil
 }
 
+// yamlUnmarshal is a wrapper around yaml.Unmarshal to ensure that the right
+// yaml package is in use. The verification for this is done using a test.
```

Comment:
please explain (to the future reader) what behavior is desired (and what is undesired) from this function.